### PR TITLE
Add polymer-lint to linters.

### PR DESCRIPTION
**Pull request by eheikes**
This PR adds linting for Polymer components.

So now we can run something like `ux-lint src/**/*.js src/**/*.html` and it will lint all of those files.

## Notes

* Files are only linted if they have an `.html` extension and have `<dom-module>` element.
* The ESLint linter was modified to only check JS files.
* By default all polymer-lint rules are used. This can be modified with a `polymer` property in the config:

    ```json
    {
        "eslint": {
            // ESLint config
        },
        "polymer": {
            "rules": {
                "style-inside-template": true
            }
        }
    }
    ```

    An empty `rules` set is equivalent to "no rules":

    ```json
    {
        "polymer": {
            "rules": {
                // no checks will take place
            }
        }
    }
    ```

## Testing

* `npm run lint` and `npm test` should not have any errors.
* You can run the linter against the Banno Online project and the output should be the same as `gulp polymer-lint`: `cd /path/to/banno-web && /path/to/ux-lint/cli.js src/**/*.html`.

### CODE REVIEW

File: linters/polymer.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const extend = require('extend');
+const filterErrors = require('polymer-lint/lib/util').filterErrors;
```
Comment (jrunning) [inline linters/polymer.js]: FWIW `const filterErrors = require('polymer-lint/lib/util/filterErrors');` would work as well.

File: linters/polymer.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const extend = require('extend');
```
Comment (jrunning) [inline linters/polymer.js]: Are we running this in any environments that don't have `Object.assign`?

Comment (eheikes) [inline linters/polymer.js]: At this point, probably not. This was a legacy thing from before Node v4. I can fix it up.

**Comment by jrunning** [comment]
Apart from the above tiny nits, this LGTM. I'm happy someone other than me was able to figure out the polymer-lint API! `busFactor++`.

**Comment by eheikes** [comment]
I fixed up the code a bit for Jordan's suggestions.

**Pull request merged**